packages/adapter/src/chains/oak.ts

```diff
@@ -211,4 +212,226 @@ export class OakAdapter extends ChainAdapter {
+   * @param collator
+   * @param amount
+   * @param percentage, an integer between 0 and 100
+   * @param keyringPair
+   * @returns
+   */
+  async delegateWithAutoCompound(
+    collator: HexString,
+    amount: BN,
+    percentage: number,
+    keyringPair: KeyringPair,
+  ): Promise<SendExtrinsicResult> {
+    const api = this.getApi();
+    this.ensureBalance(u8aToHex(keyringPair.addressRaw), amount);
+    // Check if delegation exists
+    const delegatorWalletAddress = u8aToHex(keyringPair.addressRaw);
+    const delegatorState = await this.getDelegatorState(delegatorWalletAddress);
+    let delegationsLength = 0;
+    if (!_.isUndefined(delegatorState)) {
+      const { delegations } = delegatorState;
+      const foundDelegation = _.find(
+        delegations,
+        ({ owner }) => owner.toHex() === collator,
+      );
+      delegationsLength = delegations.length;
+      if (!_.isUndefined(foundDelegation)) {
+        throw new Error("Delegation already exists");
+      }
+    }
+
+    const { minDelegation: minDelegationCodec } = api.consts.parachainStaking;
+    const minDelegation = minDelegationCodec as u128;
+    if (amount.lt(minDelegation)) {
+      throw new Error(
+        `Amount must be greater than or equal to ${minDelegation}`,
+      );
+    }
+
+    const candidateInfoCodec =
+      await api.query.parachainStaking.candidateInfo(collator);
+    const candidateInfo = candidateInfoCodec as unknown as Option<any>;
+    if (candidateInfo.isNone) {
+      throw new Error("Candidate info not found");
+    }
+    const { delegationCount: candidateDelegationCount } =
+      candidateInfo.unwrap();
+
+    console.log("collator: ", collator);
+
+    const autoCompoundingDelegationsLength =
+      await this.getAutoCompoundingDelegationsLength(collator);
+    console.log(
+      "autoCompoundingDelegationsLength: ",
+      autoCompoundingDelegationsLength,
+    );
+
+    // Delegate to collator
+    const delegateExtrinsic = api.tx.parachainStaking.delegateWithAutoCompound(
+      collator,
+      amount,
+      percentage,
+      candidateDelegationCount,
+      autoCompoundingDelegationsLength,
+      delegationsLength,
+    );
+
+    const result = await sendExtrinsic(api, delegateExtrinsic, keyringPair);
+    return result;
+  }
+
+  /**
+   * Get auto-compounding delegation percentage
+   * @param delegator
+   * @param collator
+   * @returns
+   */
+  async getAutoCompoundingDelegationPercentage(
+    collator: HexString,
+    delegator: HexString,
+  ): Promise<number> {
+    const api = this.getApi();
+    const autoCompoundingDelegationsCodec =
+      await api.query.parachainStaking.autoCompoundingDelegations(collator);
+    const autoCompoundingDelegations =
+      autoCompoundingDelegationsCodec as unknown as any[];
+    const delegation = _.find(
+      autoCompoundingDelegations,
+      (item) => item.delegator.toHex() === delegator,
+    );
+    return _.isUndefined(delegation) ? 0 : delegation.value.toNumber();
```

Comment:
This should return `undefined` but not a 0, meaning there’s no compound setup.

Return `return delegation?.value?.toNumber()` could work.

Reply:
Fixed